app/ImageConversion.hs

```diff
@@ -1,8 +1,81 @@
 module ImageConversion
-    (createImageFile, removeImage) where
+    (getGraphImage, getTimetableImage, createPDF, createImageFile, removeImage) where
 
 import System.Process
 import GHC.IO.Handle.Types
+import TimetableImageCreator (renderTable)
+import qualified Data.Map as M
+import System.Random
+import Svg.Generator
+import Database.Tables (GraphId)
+
+-- | Creates an image, and returns the name of the svg used to create the
+-- image and the name of the image
+getGraphImage :: GraphId -> M.Map String String -> IO (String, String)
+getGraphImage gId courseMap = do
+    gen <- newStdGen
+    let (rand, _) = next gen
+        svgFilename = show rand ++ ".svg"
+        imageFilename = show rand ++ ".png"
+    buildSVG gId courseMap svgFilename True
+    createImageFile svgFilename imageFilename
+    return (svgFilename, imageFilename)
+
+-- | Creates an image, and returns the name of the svg used to create the
+-- image and the name of the image
+getTimetableImage :: String -> String -> IO (String, String)
+getTimetableImage courses session = do
+    gen <- newStdGen
+    let (rand, _) = next gen
+        svgFilename = show rand ++ ".svg"
+        imageFilename = show rand ++ ".png"
+    renderTable svgFilename courses session
+    createImageFile svgFilename imageFilename
+    return (svgFilename, imageFilename)
+
+-- | Opens a new process to create a .pdf file from a .tex file and waits for that process to terminate.
+createPDF :: String -> String -> IO ()
+createPDF texName imgName = do
+  -- Get the pid of process that is opened for converting .tex to .pdf
+  (_, _, _, pid) <- convertTexToPDF texName imgName
+  print "Waiting for a process..."
+  -- Force current process to wait for conversion to finish
+  waitForProcess pid
+  print "Process Complete"
+
+-- | Create a process to use the pdflatex program to create a .pdf file from a .tex file using a supplied image
+convertTexToPDF :: String -> String -> IO
+                                   (Maybe Handle,
+                                    Maybe Handle,
+                                    Maybe Handle,
+                                    ProcessHandle)
+
+convertTexToPDF texName imgName = createProcess $ CreateProcess
+                      (ShellCommand $ "pdflatex " ++
+                              "--jobname=" ++
```

Comment:
The `pdflatex` arguments I had previously been using resulted in a .pdf with the name "export.pdf" (when using the .tex file "export.tex"). The `--jobname` argument forces the resulting .pdf to have the same name as the image it contains (a random number). I learned about this by searching how to influence the name of the .pdf resulting from a call to `pdflatex` and found [this](http://tex.stackexchange.com/questions/19182/how-to-influence-the-name-of-the-pdf-file-created-with-pdflatex-from-within-the). I thought having a random name as opposed to a constant name might be the desired result, but maybe I was wrong?